internal/app/api.go

```diff
@@ -709,6 +709,19 @@ func fieldViewsFromDesc(fds protoreflect.FieldDescriptors, isOneof bool, cd *cyc
 	return fields, nil
 }
 
+func (a *api) RetryConnection() {
+	state := a.client.conn.GetState()
+	if state == connectivity.TransientFailure || state == connectivity.Shutdown {
+		// State is currently disconnected. Do a quick retry in case the server restarted recently.
+		a.client.conn.ResetConnectBackoff()
+		stateChanged := make(chan bool)
```

Comment:
Not a major thing, but more an FYI:
Rather than using a `bool` (that uses an allocation) you can use an empty struct ie `struct{}` which uses zero allocations and then do `stateChanged <-struct{}{}`

Micro optimization. No need to change here, just thought you would like to know.

Reply:
I'm pretty new to Go so that's definitely helpful, thanks!